src/lib/swapper/swappers/ThorchainSwapper/utils/makeSwapMemo/assertIsValidMemo.ts

```diff
@@ -70,6 +74,34 @@ export const assertIsValidMemo = (memo: string): void => {
     })
   }
 
+  // Check if limit is a valid number
+  if (isNaN(parseFloat(limit)) || !isFinite(parseFloat(limit))) {
+    throw new SwapError(`[makeSwapMemo] - limit ${limit} is not a valid number`, {
+      code: SwapErrorType.MAKE_MEMO_FAILED,
+    })
+  }
+
+  // Check if affiliate is "ss"
+  if (affiliate !== THORCHAIN_AFFILIATE_NAME) {
+    throw new SwapError(
+      `[makeSwapMemo] - affiliate ${affiliate} is not ${THORCHAIN_AFFILIATE_NAME}`,
+      {
+        code: SwapErrorType.MAKE_MEMO_FAILED,
+      },
+    )
+  }
+
+  // Check if affiliateBps is a number between and including 0 and 1000 (the valid range for THORSwap)
+  const affiliateBpsNum = parseFloat(affiliateBps)
+  if (isNaN(affiliateBpsNum) || affiliateBpsNum < 0 || affiliateBpsNum > 1000) {
```

Comment:
Ditto, this triggers ample-sized JS PTSD

Reply:
Likewise, used `bn` in https://github.com/shapeshift/web/pull/4353/commits/ed83875e0eabedd9af6e6b58476170cb0e63bb5e.